lib/mumukit/auth/user.rb

```diff
@@ -40,6 +40,16 @@ def apps
   end
 
   def client
+    self.class.client
+  end
+
+  def self.from_email(email)
+    user = client.users("email:#{email}").first
+    raise Mumukit::Auth::EmailNotRegistered.new('There is no user registered with that email.') unless user.present?
+    new(user['user_id'])
+  end
+
```

Comment:
The problem is, the data retrieve from auth0 is quite different in both
requests, for example, in the users request they do not send metadata data.

El sáb., 14 de may. de 2016 a la(s) 17:23, Franco Leonardo Bulgarelli <
notifications@github.com> escribió:

> In lib/mumukit/auth/user.rb
> <https://github.com/mumuki/mumukit-auth/pull/8#discussion_r63281480>:
>
> > @@ -40,6 +40,16 @@ def apps
> >    end
> >
> >    def client
> > +    self.class.client
> > +  end
> > +
> > +  def self.from_email(email)
> > +    user = client.users("email:#{email}").first
> > +    raise Mumukit::Auth::EmailNotRegistered.new('There is no user registered with that email.') unless user.present?
> > +    new(user['user_id'])
> > +  end
> > +
>
> Aren't you making two requests? Since you added the extra user param to
> the constructor, you here pass the user object
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/mumuki/mumukit-auth/pull/8/files/bf45e2d240cd8d6689aaa361db9e155887c0583c#r63281480>
>